### PR TITLE
Rod cutting problem in python

### LINKED ISSUE #2997: Rod Cutting Problem [DP] [Python]
## 🚀 Feature
There is no problem with ROD cutting questions based on Python programming. I would like to contribute to it.
No Rod cutting problem in DP python.
I want to assign to this issue.

### PR DESCRIPTION
**Pull request by sakar97**
Fix #2997 

This is a very famous dynamic programming question where the problem statement is as follows:
**Given a rod of length n inches and an array of prices that contains prices of all pieces of size smaller than n.
   Determine the maximum value obtainable by cutting up the rod and selling the pieces.**

`TC : O(N^2)`

### CODE REVIEW

File: Python/dp/Rod_cutting_dp.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+# Driver program to test above functions
+
+
+if __name__ == "__main__":
+
+    arr = [1, 5, 8, 9, 10, 17, 17, 20]
+    size = len(arr)
+
+    # User input for price array and length of the rod.
```
Comment (gat786) [inline Python/dp/Rod_cutting_dp.py:46]: where is the input taken?

Comment (gat786) [reply Python/dp/Rod_cutting_dp.py:46]: As far as I can see you are passing the size of the array as n to the cutRod function

File: Python/dp/Rod_cutting_dp.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    Time Complexity: O(N^2)
+'''
+# A Dynamic Programming solution for Rod cutting problem
+INT_MIN = -32767
+
+# Returns the best obtainable price for a rod of length n and
+# price[] as prices of different pieces
+
+
+def cutRod(price, n):
```
Comment (gat786) [inline Python/dp/Rod_cutting_dp.py:23]: what does the variable `n` mean here? It be better if the variable name is something which gives a context of what it is supposed to store

File: Python/dp/Rod_cutting_dp.py (hunk 1 above)
Comment (sakar97) [reply Python/dp/Rod_cutting_dp.py:46]: arr[] is the price array whereas the size of that array is N. The rod is the same length as the price array. Everything is explained in the code DOC. Thanks

File: Python/dp/Rod_cutting_dp.py (hunk 2 above)
Comment (sakar97) [reply Python/dp/Rod_cutting_dp.py:23]: N is simply the size of the rod.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Make README changes as well

File: Python/dp/Rod_cutting_dp.py (hunk 2 above)
Comment (gat786) [reply Python/dp/Rod_cutting_dp.py:23]: Yeah i meant you should change it

File: Python/dp/Rod_cutting_dp.py (hunk 1 above)
Comment (gat786) [reply Python/dp/Rod_cutting_dp.py:46]: Yeah well can some one put in dynamic rod size? If it is possible then make the entry user interactable.

**Comment by atarax665** [comment]
No update

**Pull request closed without merging**